src/index.ts

```diff
@@ -119,12 +119,17 @@ export const htmlPlugin = (configuration: Configuration = { files: [], }): esbui
         }
     }
 
-    function renderTemplate(htmlFileConfiguration: HtmlFileConfiguration, htmlTemplate: string) {
-        const templateContext = {
-            define: htmlFileConfiguration.define,
+    async function renderTemplate({ htmlTemplate, define }: HtmlFileConfiguration, log: boolean) {
+        const templateContext = { define }
+        const template = (htmlTemplate && fs.existsSync(htmlTemplate)
+            ? await fs.promises.readFile(htmlTemplate)
+            : htmlTemplate || '').toString()
+        const isValid = template.startsWith('<!DOCTYPE html>')
```

Comment:
I feel like this is a bit too error prone. For example, the templates in the README don't work with this anymore. Can we leave this out for now? We can create a issue for checking the validity of a document in the future. :)